lib/spack/spack/fetch_strategy.py

```diff
@@ -990,3 +1072,11 @@ def __init__(self, method):
         super(NoStageError, self).__init__(
             "Must call FetchStrategy.set_stage() before calling %s" %
             method.__name__)
+
+
+class PyPiFetchError(FetchError):
```

Comment:
This class is only instantiated once.  Unless we explicitly check for it elsewhere, we don't really need this.  Just put `raise FetchError(...)` where you have `raise PyPiFetchError(...)` above.